events/codepipeline.go

```diff
@@ -0,0 +1,106 @@
+package events
+
+import (
+	"time"
+)
+
+const (
+	CodePipelineEventSource              = "aws.codepipeline"
+	CodePipelineExecutionEventDetailType = "CodePipeline Pipeline Execution State Change"
+	CodePipelineActionEventDetailType    = "CodePipeline Action Execution State Change"
+	CodePipelineStageEventDetailType     = "CodePipeline Stage Execution State Change"
+)
+
+type CodePipelineStageState string
+
+const (
+	CodePipelineStageStateStarted   CodePipelineStageState = "STARTED"
+	CodePipelineStageStateSucceeded                        = "SUCCEEDED"
+	CodePipelineStageStateResumed                          = "RESUMED"
+	CodePipelineStageStateFailed                           = "FAILED"
+	CodePipelineStageStateCanceled                         = "CANCELED"
+)
+
+type CodePipelineState string
+
+const (
+	CodePipelineStateStarted    CodePipelineState = "STARTED"
+	CodePipelineStateSucceeded                    = "SUCCEEDED"
+	CodePipelineStateResumed                      = "RESUMED"
+	CodePipelineStateFailed                       = "FAILED"
+	CodePipelineStateCanceled                     = "CANCELED"
+	CodePipelineStateSuperseded                   = "SUPERSEDED"
+)
+
+type CodePipelineActionState string
+
+const (
+	CodePipelineActionStateStarted   CodePipelineActionState = "STARTED"
+	CodePipelineActionStateSucceeded                         = "SUCCEEDED"
+	CodePipelineActionStateFailed                            = "FAILED"
+	CodePipelineActionStateCanceled                          = "CANCELED"
+)
+
+// CodePipelineEvent is documented at:
+// https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/EventTypes.html#codepipeline_event_type
+type CodePipelineEvent struct {
```

Comment:
Suggest changing the name to `CodePipelineCloudWatchEvent` to disambiguate from `CodePipelineJobEvent`